lib/info.gi

```diff
@@ -32,34 +32,36 @@
 ##
 #R  IsInfoClassListRep(<obj>)                       length one positional rep
 ##                              
-##  An InfoClass is represented as a length one positional object with
-##  a positive integer in ic![1]. No other representations are 
-##  anticipated
+##  An InfoClass is represented as a positional object with the following
+##  members:
 ##
-
-DeclareRepresentation("IsInfoClassListRep", IsPositionalObjectRep,[]);
-
-#############################################################################
-##
-#V  InfoData                 record of private stuff
-#V  InfoData.CurrentLevels   the current desired verbosity levels set by user
-#V  InfoData.ClassNames      names of all info classes, used for printing
-#V  InfoData.Handler         optional handler for InfoDoPrint
+##  1 : Current level (A positive integer)
+##  2 : ClassName     (String)
+##  3 : Handler       (Optional, handler for InfoDoPrint)
+##  4 : Output        (Optional, output stream)
+##  5 : ClassNum      (An integer identifying the class)
 ##
-##  these are all lists, and the first two should have the same length, 
-##  which defines the number of Info classes that exist
+##  ClassNum is placed at the end so the length of the list never changes
 ##
 
-if not IsBound(InfoData) then
+DeclareRepresentation("IsInfoClassListRep", IsAtomicPositionalObjectRep,[]);
 
-    BIND_GLOBAL( "InfoData", rec() );
-    InfoData.CurrentLevels := [];
-    InfoData.ClassNames := [];
-    InfoData.Handler := [];
-    InfoData.Output := [];
-    if IsBound(HPCGAP) then
-        ShareInternalObj(InfoData);
-    fi;
+# Give constants for the members of info class
+BIND_CONSTANT("INFODATA_CURRENTLEVEL", 1);
+BIND_CONSTANT("INFODATA_CLASSNAME", 2);
+BIND_CONSTANT("INFODATA_HANDLER", 3);
+BIND_CONSTANT("INFODATA_OUTPUT", 4);
+BIND_CONSTANT("INFODATA_NUM", 5);
+
+# A list of all created Info Classes
+INFO_CLASSES := [];
+
+# The most recently called Info statement
+InfoData := rec();
```

Comment:
I actually think that we can get rid of `InfoData` but that can be done in a later PR (I can take care of it, too).

Reply:
In the meantime, perhaps say something like this:
> The level and (first) selector of the most recently executed Info statement is stored in InfoData

Reply:
I will improve the comment. Note that `InfoData` is used by GAPDoc. I tried hard in this commit to not break any way someone might have extended the `Info`, in case anyone has.